blogs/2022/09/15/dev-container-features.md

```diff
@@ -0,0 +1,141 @@
+        "version": "latest",
+        "moby": true
+    }
+}
+```
+
+You can also explore official and publicly contributed Features [on the spec site](https://containers.dev/features). Any Feature can be added by editing `devcontainer.json`, and publicly published ones can be added through existing dev container configuration experiences (like what you’d see in the [VS Code Remote - Containers extension](https://code.visualstudio.com/docs/remote/containers#_dev-container-features-preview)).
+
+![Spec site list of available Features](./features-list-site.png)
+
+You can even use dev containers with Features from your favorite CI system using the [dev container CLI](https://github.com/devcontainers/cli), a GitHub Action, or an Azure DevOps tasks. We have a GitHub Action and Azure DevOps task available in the [devcontainers/ci repo](https://github.com/devcontainers/ci). The dev container CLI, GitHub Action, or Azure DevOps task can also be used to [pre-build images](https://code.visualstudio.com/docs/remote/devcontainer-cli#_prebuilding) that include Feature contents to speed up start times.
+
+If you’d like to not only use publicly available Features but also create your own private or public ones to share, continue reading!
+
+## Authoring
+A great place to get started with creating your own Features is the new [Features template repository](https://github.com/devcontainers/feature-template).  Beyond including a good template for the contents of a given Feature, the template also includes a GitHub Actions workflow to quickly publish them too, using the [GitHub Container Registry](https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-container-registry) (GHCR) for your account to get you up and running as fast as possible. We’ll talk more about publishing in a minute.
+
+The source code of a Feature has two components: An install script (`install.sh`) and a configuration file (`devcontainer-feature.json`).
+
+```
++-- feature
+|    +-- devcontainer-feature.json
+|    +-- install.sh
+|    +-- (other files)
+
+```
+
+`install.sh`: The installation entry point script – it’s conceptually added as a layer of the image’s Dockerfile and executed during build-time. This entry point script can install tools such as languages (ruby) and tools (GitHub CLI).
+
+`devcontainer-feature.json`: This contains metadata about the Feature, a set of options that can be passed to a Feature’s install script during installation, as well as "pieces" of devcontainer.json that will be merged into the final dev container. For example, if any Feature indicates `privileged`: true in their config, the entire dev container will be started with the `--privileged` flag.
+
+Features can be authored in a variety of languages, the most straightforward being shell scripts. If a Feature is authored in a different language, information about it should be included in the metadata so that users can make an informed choice about it.
```

Comment:
I am not sure if this is a little misleading.  While the install.sh will run anything, if you wrote a feature in an interpreted language that wasn't present on the build machine, the code would fail to execute.

Reply:
I'll add what you mention to the note below, i.e.:

> **Note:** While the install.sh will run anything, if you wrote a feature in an interpreted language that wasn't present on the build machine, the code would fail to execute...

Reply:
@joshspicer It's not so much the build machine as it is the container isn't it?  Install.sh runs during the build not locally.  So the thing to get across is if you use a node scruot, you need to be sure to acquire node as a part of your install.sh script.

Reply:
Updated to:

> Note: While install.sh will run Features in any language, if you wrote a Feature in an interpreted language that wasn't present in the dev container, the code would fail to execute. Be sure to acquire the language(s) you need as part of install.sh.